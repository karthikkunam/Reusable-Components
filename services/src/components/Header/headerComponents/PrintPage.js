import React from 'react';

export default class PrintPage extends React.PureComponent {
  PrintPage = e => {
    e.preventDefault();
    window.print();
  };
  render() {
    return <a className="nav-item-icon icon icon-print" href="#" onClick={this.PrintPage} />;
  }
}
