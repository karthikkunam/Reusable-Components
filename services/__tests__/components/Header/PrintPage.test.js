import React from 'react';
import PrintPage from '../../../src/components/Header/headerComponents/PrintPage';
import { shallow } from 'enzyme';

describe('PrintPage Component', () => {
  it('renders correctly', () => {
    const component = shallow(<PrintPage />);
    component.find('a').simulate('click', {
      preventDefault: () => {},
    });
    expect(component).toMatchSnapshot();
  });
  it('mocks and calls window.print', () => {
    const component = shallow(<PrintPage />);
    window.print = jest.fn();
    window.print();
    expect(window.print).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
