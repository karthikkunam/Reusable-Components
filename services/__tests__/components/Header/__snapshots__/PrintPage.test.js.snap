// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintPage Component mocks and calls window.print 1`] = `
<a
  className="nav-item-icon icon icon-print"
  href="#"
  onClick={[Function]}
/>
`;

exports[`PrintPage Component renders correctly 1`] = `
<a
  className="nav-item-icon icon icon-print"
  href="#"
  onClick={[Function]}
/>
`;
